docs/source/routing/query-planning/query-planning-best-practices.mdx

```diff
@@ -0,0 +1,181 @@
+---
+title: Query Planning Best Practices
+subtitle: What things should we keep in mind as we construct our supergraph and the operations that consume them?
+---
+
+When working with Apollo Federation, changes in our schema can have unexpected impact on the complexity of our graph.
+While many issues can be found at build time with [schema composition](https://www.apollographql.com/docs/graphos/schema-design/federated-schemas/composition), others changes may lead to issues that only arise at runtime while generating or executing query plans.
+
+Examples of changes that can impact query planning include:
+* Modifying `@key`, `@requires`, `@provides`, or `@shareable` directive usage
+* Adding or removing a type implementation from an interface
+* Using `interfaceObject` and adding new fields to an interface
+
+To help alleviate these issues as much as possible, we recommend following some of these best practices for your federated graph.
+
+## Shared Types and Fields
+
+The [`@shareable` directive](https://www.apollographql.com/docs/graphos/schema-design/federated-schemas/sharing-types) allows multiple subgraphs to resolve the same types or fields on entities, giving the query planner options for potentially shorter query paths. However, it's important to use it judiciously.
+- Extensive use can exponentially increase the number of possible query plans generated, potentially leading to performance degradation at runtime as we generate uncached plans.
+- Using `@shareable` at root fields on the `Query`, `Mutation`, and `Subscription` types indicates that any subgraph can resolve a given entry point. While query plans can be deterministic for a given version of Router + Federation, there are no guarantees across versions, meaning that your plans may change if new services get added or deleted. This could cause an unexpected change in traffic a service even there were no changes in the operations.
```

Comment:
It's worth caveating that if a root query returns a list, the router may (and is likely to) run a query in parallel and expects the results to be in the _exact same order_.